src/node/nodetonode.h

```diff
@@ -89,23 +117,44 @@ namespace ccf
       return t;
     }
 
+    template <class T>
+    std::vector<uint8_t> recv_authenticated_with_load(
+      const uint8_t*& data, size_t& size)
+    {
+      // data contains the message header of type T, the raw data, and the gcm
+      // header at the end
+      const auto& t = serialized::peek<T>(data, size);
+      const auto d = serialized::read(data, size, (size - sizeof(GcmHdr)));
```

Comment:
As discussed, we should be able to avoid the copy of the payload altogether here (as well as the subsequent skipping of `PbftHeader` after calling this function from `pbft,h`).